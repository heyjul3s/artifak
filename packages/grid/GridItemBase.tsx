import { createBaseComponent } from '@artifak/block';

import {
  gridArea,
  gridColumn,
  gridRow,
  justifySelf,
  alignSelf,
  AlignSelfProps,
  GridAreaProps,
  GridColumnProps,
  GridRowProps,
  JustifySelfProps,
} from 'styled-system';

export const GridItemBase = createBaseComponent<
  | AlignSelfProps
  | GridAreaProps
  | GridRowProps
  | GridColumnProps
  | JustifySelfProps
>({}, [alignSelf, gridArea, gridRow, gridColumn, justifySelf]);
