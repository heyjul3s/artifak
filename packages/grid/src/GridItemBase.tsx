import { createStyledComponent } from '@artifak/component-generator';

import {
  gridArea,
  gridColumn,
  gridRow,
  justifySelf,
  alignSelf
} from 'styled-system';

import { GridItemBaseProps } from './typings';

export const GridItemBase = createStyledComponent<GridItemBaseProps>({}, {}, [
  alignSelf,
  gridArea,
  gridRow,
  gridColumn,
  justifySelf
]);
