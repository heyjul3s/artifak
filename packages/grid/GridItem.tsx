import React, { forwardRef } from 'react';
import { GridItemBase } from './GridItemBase';

export const GridItem = forwardRef((props: any, ref) => (
  <GridItemBase {...props} ref={ref}>
    {props.children}
  </GridItemBase>
));
