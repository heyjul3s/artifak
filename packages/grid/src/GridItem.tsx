import React from 'react';
import { GridItemBase } from './GridItemBase';

/* eslint-disable @typescript-eslint/no-explicit-any */
export const GridItem = React.forwardRef((props: any, ref) => (
  <GridItemBase {...props} ref={ref}>
    {props.children}
  </GridItemBase>
));
