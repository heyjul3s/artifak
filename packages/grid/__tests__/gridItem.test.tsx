import React from 'react';
import { GridItem } from '../GridItem';
import { GridItemBase } from '../GridItemBase';
import renderer from 'react-test-renderer';

describe('GridItemBase', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<GridItemBase>Hello World</GridItemBase>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('GridItem', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<GridItem>Hello World</GridItem>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
