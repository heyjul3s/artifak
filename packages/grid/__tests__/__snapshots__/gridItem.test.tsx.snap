// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem should render correctly 1`] = `
<div
  className="sc-AxirZ ktnnZK"
>
  Hello World
</div>
`;

exports[`GridItemBase should render correctly 1`] = `
<div
  className="sc-AxirZ ktnnZK"
>
  Hello World
</div>
`;
